lib/spreadsheet_exporter/xlsx.rb

```diff
@@ -1,43 +1,182 @@
 require 'write_xlsx'
 require_relative 'spreadsheet'
+require "active_support"
+require "active_support/core_ext/hash/keys"
 
 module SpreadsheetExporter
   module XLSX
-    def self.from_objects(objects, options = {})
-      spreadsheet = Spreadsheet.from_objects(objects, options).compact
-      from_spreadsheet(spreadsheet)
+    extend Writexlsx::Utility # gets us `xl_rowcol_to_cell` and `xl_col_to_name`
+
+    ROW_MAX = 65_536 - 1
+    DATA_WORKSHEET_NAME = "data".freeze
+
+    def self.from_objects(objects, humanize_headers_class: nil, **options)
+      spreadsheet = Spreadsheet.from_objects(objects, humanize_headers_class: humanize_headers_class, **options).compact
+      from_spreadsheet(spreadsheet, **options)
     end
 
-    def self.from_spreadsheet(spreadsheet, temp_file_path = 'tmp/items.xlsx')
-      # Create a new Excel workbook
-      workbook = WriteXLSX.new(temp_file_path)
+    def self.from_spreadsheet(spreadsheet, validations: {}, data_sources: {}, freeze_panes: false, **options)
```

Comment:
This is technically a breaking change since `temp_file_path` is not accepted anymore, but it doesn't affect any of our apps